src/store_swapout.cc

```diff
@@ -304,8 +304,9 @@ storeSwapOutFileClosed(void *data, int errflag, StoreIOState::Pointer self)
             storeConfigure();
         }
 
-        e->disk().finalizeSwapoutFailure(*e);
         e->releaseRequest(); // TODO: Keep the memory entry (if any)
+        e->swap_status = SWAPOUT_FAILED;
+        e->disk().finalizeSwapoutFailure(*e);
```

Comment:
This change invalidates the comment in Fs::Ufs::UFSSwapDir::finalizeSwapoutFailure(). I am not sure whether adjusting the comment alone is sufficient (or more work is required to bring everything in sync).